flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -540,39 +571,41 @@ private FinalApplicationStatus getYarnStatus(ApplicationStatus status) {
 	/**
 	 * Request new container if pending containers cannot satisfy pending slot requests.
 	 */
-	private void requestYarnContainerIfRequired() {
-		int requiredTaskManagers = getNumberRequiredTaskManagers();
-
-		if (requiredTaskManagers > numPendingContainerRequests) {
-			requestYarnContainer();
-		}
+	private void requestYarnContainerIfRequired(Resource containerResource) {
```

Comment:
Shouldn't we iterate over all resources which are needed instead of restricting it to `containerResource`?